conn.go

```diff
@@ -403,9 +403,14 @@ func (c *Conn) closeWithError(err error) {
 	}
 
 	// if error was nil then unblock the quit channel
-	close(c.quit)
+	if err == nil {
+		close(c.quit)
```

Comment:
Im not sure this is correct, as were stopping this will cause the `writeToConn` goroutine to leak